graviti/dataframe/frame.py

```diff
@@ -67,31 +68,36 @@ class DataFrame:
 
     def __init__(
         self,
-        data: Union[Sequence[Sequence[Any]], Dict[str, Any], "DataFrame", "LazyLists", None] = None,
+        data: Union[Iterable[NestedDict[str, Any]], "DataFrame", None] = None,
         schema: Any = None,
         columns: Optional[Iterable[str]] = None,
     ) -> None:
         if data is None:
-            data = {}  # type: ignore[assignment]
-            # https://github.com/python/mypy/issues/6463
+            data = {}
         if schema is not None:
             # TODO: missing schema processing
             pass
         if columns is not None:
             # TODO: missing columns processing
             pass
 
+        if not isinstance(data, DataFrame) and isinstance(data, Iterable):
+            self._init(pa.array(data))
+        else:
+            raise ValueError("DataFrame only supports generating from Sequence now")
+
+    def _init(self, array: pa.StructArray) -> None:
         self._columns = {}
         self._column_names = []
-        if isinstance(data, dict):
-            for key, value in data.items():
-                self._columns[key] = (
-                    DataFrame(value) if isinstance(value, dict) else ColumnSeries(value, name=key)
-                )
-                self._column_names.append(key)
-            self._index = ColumnSeries(list(range(self.__len__())))
-        else:
-            raise ValueError("DataFrame only supports generating from dictionary now")
+        for key in array[0].keys():
```

Comment:
why we need to get the first element of array?
what if the array is an empty array?